diff -urN gobject-introspection-1.56.1.orig/Makefile.in gobject-introspection-1.56.1/Makefile.in
--- gobject-introspection-1.56.1.orig/Makefile.in	2018-05-24 17:41:04.000000000 +0100
+++ gobject-introspection-1.56.1/Makefile.in	2018-05-24 17:50:28.000000000 +0100
@@ -991,7 +991,7 @@
         PYTHONPATH=$(top_builddir):$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_BUILDDIR=$(top_builddir) \
-        $(top_builddir)/g-ir-scanner
+        $(PYTHON) $(top_builddir)/g-ir-scanner
 
 INTROSPECTION_SCANNER_ARGS = \
     --verbose \
@@ -1024,7 +1024,7 @@
         PYTHONPATH=$(top_builddir):$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_BUILDDIR=$(top_builddir) \
-        $(top_builddir)/g-ir-doc-tool
+        $(PYTHON) $(top_builddir)/g-ir-doc-tool
 
 INTROSPECTION_DOCTOOL_ARGS = \
     --add-include-path=$(srcdir) \
diff -urN gobject-introspection-1.56.1.orig/common.mk gobject-introspection-1.56.1/common.mk
--- gobject-introspection-1.56.1.orig/common.mk	2018-05-24 17:41:04.000000000 +0100
+++ gobject-introspection-1.56.1/common.mk	2018-05-24 17:45:03.000000000 +0100
@@ -13,7 +13,7 @@
         PYTHONPATH=$(top_builddir):$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_BUILDDIR=$(top_builddir) \
-        $(top_builddir)/g-ir-scanner
+        $(PYTHON) $(top_builddir)/g-ir-scanner
 
 INTROSPECTION_SCANNER_ARGS = \
     --verbose \
@@ -45,7 +45,7 @@
         PYTHONPATH=$(top_builddir):$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_BUILDDIR=$(top_builddir) \
-        $(top_builddir)/g-ir-doc-tool
+        $(PYTHON) $(top_builddir)/g-ir-doc-tool
 
 INTROSPECTION_DOCTOOL_ARGS = \
     --add-include-path=$(srcdir) \
diff -urN gobject-introspection-1.56.1.orig/tests/Makefile.in gobject-introspection-1.56.1/tests/Makefile.in
--- gobject-introspection-1.56.1.orig/tests/Makefile.in	2018-05-24 17:41:04.000000000 +0100
+++ gobject-introspection-1.56.1/tests/Makefile.in	2018-05-24 17:49:50.000000000 +0100
@@ -620,7 +620,7 @@
         PYTHONPATH=$(top_builddir):$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_BUILDDIR=$(top_builddir) \
-        $(top_builddir)/g-ir-scanner
+        $(PYTHON) $(top_builddir)/g-ir-scanner
 
 INTROSPECTION_SCANNER_ARGS = \
     --verbose \
@@ -653,7 +653,7 @@
         PYTHONPATH=$(top_builddir):$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_BUILDDIR=$(top_builddir) \
-        $(top_builddir)/g-ir-doc-tool
+        $(PYTHON) $(top_builddir)/g-ir-doc-tool
 
 INTROSPECTION_DOCTOOL_ARGS = \
     --add-include-path=$(srcdir) \
diff -urN gobject-introspection-1.56.1.orig/tests/offsets/Makefile.in gobject-introspection-1.56.1/tests/offsets/Makefile.in
--- gobject-introspection-1.56.1.orig/tests/offsets/Makefile.in	2018-05-24 17:41:04.000000000 +0100
+++ gobject-introspection-1.56.1/tests/offsets/Makefile.in	2018-05-24 17:47:11.000000000 +0100
@@ -621,7 +621,7 @@
         PYTHONPATH=$(top_builddir):$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_BUILDDIR=$(top_builddir) \
-        $(top_builddir)/g-ir-scanner
+        $(PYTHON) $(top_builddir)/g-ir-scanner
 
 INTROSPECTION_SCANNER_ARGS = \
     --verbose \
@@ -654,7 +654,7 @@
         PYTHONPATH=$(top_builddir):$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_BUILDDIR=$(top_builddir) \
-        $(top_builddir)/g-ir-doc-tool
+        $(PYTHON) $(top_builddir)/g-ir-doc-tool
 
 INTROSPECTION_DOCTOOL_ARGS = \
     --add-include-path=$(srcdir) \
diff -urN gobject-introspection-1.56.1.orig/tests/scanner/Makefile.in gobject-introspection-1.56.1/tests/scanner/Makefile.in
--- gobject-introspection-1.56.1.orig/tests/scanner/Makefile.in	2018-05-24 17:41:04.000000000 +0100
+++ gobject-introspection-1.56.1/tests/scanner/Makefile.in	2018-05-24 17:48:38.000000000 +0100
@@ -722,7 +722,7 @@
         PYTHONPATH=$(top_builddir):$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_BUILDDIR=$(top_builddir) \
-        $(top_builddir)/g-ir-scanner
+        $(PYTHON) $(top_builddir)/g-ir-scanner
 
 INTROSPECTION_SCANNER_ARGS = --verbose -I$(top_srcdir) \
 	--add-include-path=$(srcdir) \
@@ -753,7 +753,7 @@
         PYTHONPATH=$(top_builddir):$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_BUILDDIR=$(top_builddir) \
-        $(top_builddir)/g-ir-doc-tool
+        $(PYTHON) $(top_builddir)/g-ir-doc-tool
 
 INTROSPECTION_DOCTOOL_ARGS = \
     --add-include-path=$(srcdir) \
diff -urN gobject-introspection-1.56.1.orig/tests/scanner/annotationparser/Makefile.in gobject-introspection-1.56.1/tests/scanner/annotationparser/Makefile.in
--- gobject-introspection-1.56.1.orig/tests/scanner/annotationparser/Makefile.in	2018-05-24 17:41:04.000000000 +0100
+++ gobject-introspection-1.56.1/tests/scanner/annotationparser/Makefile.in	2018-05-24 17:49:14.000000000 +0100
@@ -523,7 +523,7 @@
         PYTHONPATH=$(top_builddir):$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_BUILDDIR=$(top_builddir) \
-        $(top_builddir)/g-ir-scanner
+        $(PYTHON) $(top_builddir)/g-ir-scanner
 
 INTROSPECTION_SCANNER_ARGS = \
     --verbose \
@@ -556,7 +556,7 @@
         PYTHONPATH=$(top_builddir):$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_BUILDDIR=$(top_builddir) \
-        $(top_builddir)/g-ir-doc-tool
+        $(PYTHON) $(top_builddir)/g-ir-doc-tool
 
 INTROSPECTION_DOCTOOL_ARGS = \
     --add-include-path=$(srcdir) \
diff -urN gobject-introspection-1.56.1.orig/tests/warn/Makefile.in gobject-introspection-1.56.1/tests/warn/Makefile.in
--- gobject-introspection-1.56.1.orig/tests/warn/Makefile.in	2018-05-24 17:41:04.000000000 +0100
+++ gobject-introspection-1.56.1/tests/warn/Makefile.in	2018-05-24 17:46:02.000000000 +0100
@@ -522,7 +522,7 @@
         PYTHONPATH=$(top_builddir):$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_BUILDDIR=$(top_builddir) \
-        $(top_builddir)/g-ir-scanner
+        $(PYTHON) $(top_builddir)/g-ir-scanner
 
 INTROSPECTION_SCANNER_ARGS = \
     --verbose \
@@ -555,7 +555,7 @@
         PYTHONPATH=$(top_builddir):$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_BUILDDIR=$(top_builddir) \
-        $(top_builddir)/g-ir-doc-tool
+        $(PYTHON) $(top_builddir)/g-ir-doc-tool
 
 INTROSPECTION_DOCTOOL_ARGS = \
     --add-include-path=$(srcdir) \
